File: LLD/Basic/BankAccount/BankAccount.java
package BankAccount;
/*
 * It should have 3 data members
accountNumber: String
balance: int
roi:double (Should represent rate of interest)
It should have 2 methods
getSimpleInterest: It should take time (in years) as a parameter. The data type of time should be int. It should return Simple Interest as a double.
getBalanceWithInterest: It should take time (in years) as a parameter. The data type of time should be int. It should return a new balance (including simple interest) as a double.
 */
public class BankAccount {
    // write the code of bank account class here
    public String accountNumber;
    public int balance;
    public double roi;

    public BankAccount() {
        this.accountNumber = "";
        this.balance = 0;
        this.roi = 0.0;
    }
    public BankAccount(String accountNumber, int balance, double roi){
        this.accountNumber = accountNumber;
        this.balance = balance;
        this.roi = roi;
    }

    double getSimpleInterest(int time){
        return (balance*roi*time)/100.0;
    }
    double getBalanceWithInterest(int time){
        double interest = getSimpleInterest(time);
        return balance+interest;
    }

    public String getaccountNumber(){
        return accountNumber;
    }
    public void setaccountNumber(String accountNumber){
        this.accountNumber = accountNumber;
    }
    public int getBalance() {
        return balance;
    }

    public void setBalance(int balance) {
        this.balance = balance;
    }
    public double getRoi() {
        return roi;
    }

    public void setRoi(double roi) {
        this.roi = roi;
    }
}

File: LLD/Basic/BankAccount/BankAccount.py
# Bank Account Class Implementation in Python
# Problem Statement
# You are tasked with creating a Python class for managing bank accounts. The class should encapsulate the functionality required for handling account details and computing interest. Your task is to implement a Bank Account class with the following specifications:

# Requirements
# The Bank Account class should have three data members:
# accountNumber: String (representing the account number)
# balance: int (representing the current balance)
# roi: double (representing the rate of interest)
# The Bank Account class should have two methods:
# getSimpleInterest: This method should take the time (in years) as a parameter (data type: int). It should return the Simple Interest as a double.
# getBalanceWithInterest: This method should take the time (in years) as a parameter (data type: int). It should return a new balance (including simple interest) as a double.
# Instructions
# Implement the BankAccount class according to the specified requirements.
# Ensure that the class contains the specified data members and methods.
# The getSimpleInterest method should calculate and return the simple interest based on the provided time (in years) and the rate of interest (roi).
# The getBalanceWithInterest method should calculate and return the new balance after adding the simple interest to the current balance based on the provided time (in years) and the rate of interest (roi).

class BankAccount:
    def __init__(self, account_number="", balance=0, roi=0.0):
        self.account_number = account_number
        self.balance = balance
        self.roi = roi

    def getSimpleInterest(self, time):
        return (self.balance * self.roi * time) / 100.0

    def getBalanceWithInterest(self, time):
        interest = self.getSimpleInterest(time)
        return self.balance + interest

    def getaccountNumber(self):
        return self.account_number

    def setaccountNumber(self, account_number):
        self.account_number = account_number

    def getBalance(self):
        return self.balance

    def setBalance(self, balance):
        self.balance = balance

    def getRoi(self):
        return self.roi

    def setRoi(self, roi):
        self.roi = roi
